common/src/OpenSpaceNet.cpp

```diff
@@ -363,359 +400,282 @@ void OpenSpaceNet::initFilter()
-            {
-                SlidingWindowChipper chipper(item.second,
-                                             windowSizes,
-                                             resampledSize,
-                                             model_->metadata().modelSize());
-
-                Subsets subsets;
-                copy(chipper, back_inserter(subsets));
-
-                auto predictions = model_->detectBoxes(subsets);
-
-                if(!args_.excludeLabels.empty()) {
-                    std::set<string> excludeLabels(args_.excludeLabels.begin(), args_.excludeLabels.end());
-                    predictions = filterLabels(predictions, LabelFilterType::EXCLUDE, excludeLabels);
-                }
-
-                if(!args_.includeLabels.empty()) {
-                    std::set<string> includeLabels(args_.includeLabels.begin(), args_.includeLabels.end());
-                    predictions = filterLabels(predictions, LabelFilterType::INCLUDE, includeLabels);
-                }
-
-                for(auto& prediction : predictions) {
-                    prediction.window.x += item.first.x;
-                    prediction.window.y += item.first.y;
-                    addFeature((PredictionPoly)prediction);
-                }
-            }
+    DG_CHECK(!args_.resampledSize || *args_.resampledSize <= modelSize_.width,
+             "Argument --resample-size (size: %d) does not fit within the model (width: %d).",
+             *args_.resampledSize, modelSize_.width)
 
-            if(pd_ && !args_.quiet) {
-                pd_->update(classifyCategory_, (float)++curBlockClass / numBlocks);
-            }
+    if (!args_.resampledSize) {
+        for (auto c : args_.windowSize) {
+            DG_CHECK(c <= modelSize_.width,
+                     "Argument --window-size contains a size that does not fit within the model (width: %d).",
+                      modelSize_.width)
         }
-    });
-
-    consumerFuture.wait();
-    if(pd_ && !args_.quiet) {
-        pd_->stop();
     }
 
-    image_->rethrowIfError();
+    Detector::Ptr modelNode;
+    if(metadata_->category() == "segmentation") {
+        initSegmentation(model);
+        modelNode = deepcore::classification::node::PolyDetector::create("Model");
+    } else {
+        modelNode = deepcore::classification::node::BoxDetector::create("Model");
+    }
 
-    skipLine();
+    modelNode->attr("model") = model;
+    modelNode->attr("confidence") = confidence;
+    return modelNode;
 }
 
-void OpenSpaceNet::processSerialPolys()
+void OpenSpaceNet::initSegmentation(Model::Ptr model)
 {
-    auto predictions = processSerial<PredictionPoly>();
-
-    if(args_.action != Action::LANDCOVER && args_.nms) {
-        skipLine();
-        OSN_LOG(info) << "Performing non-maximum suppression..." ;
-        predictions = polygonalNonMaxSuppression(predictions, args_.overlap / 100);
-    }
+    auto castModel = dynamic_cast<CaffeSegmentation*>(model.get());
```

Comment:
No reason that we can't support other segmentation models. Also use `std::dynamic_pointer_cast`
```cpp
auto segmentation = dynamic_pointer_cast<Segmentation>(model.get());
```